daemons/fenced/fenced_history.c

```diff
@@ -28,6 +28,12 @@
 
 #define MAX_STONITH_HISTORY 500
 
+static bool
+stonith_op_state_pending(remote_fencing_op_t *op)
+{
+    return op->state != st_failed && op->state != st_done;
+}
+
```

Comment:
That was due ;-)
Question is if we shouldn't place it more globally. At least crm_mon could make use of it.
Optimization used in most places will probably automatically inline this kind of short backend-functions so that we can have it "static inline" in a header as well.

Reply:
It would be nice, but it doesn't need to be part of this PR. When someone has time they could grab it.

It could take the state directly (rather than the op) so we could put it in stonith-ng.h and be usable in libstonithd and libpacemaker as well as pacemaker-fenced (crm_mon doesn't need it directly anymore). Alternatively we could keep this in pacemaker-fenced.h for pacemaker-fenced use and add an equivalent for stonith_history_t for use by libstonithd and libpacemaker.

Reply:
Makes sense. I don't mind to make it available now so that this part can be the first consumer of it :-)

Reply:
I would opt for the first suggestion from Ken but yes  - doesn't have to happen right away.
When touching a bunch of other files it anyway should be a separate commit that can go into a separate PR.

Reply:
As long as it's a separate commit, this PR is fine. I think there are fewer than a dozen places that would use the new function.

The name is fine; libstonithd uses the stonith_ prefix currently. Eventually I want to combine some libraries together, and the guidelines might change then, but for now that's correct.

Reply:
Thought you might have a vision already we might start implementing with new functions before doing the big turn-over (as with pcmk_(_) ).

Reply:
Not clear yet ... if we end up merging it into libcrmcommon, then it would get pcmk, but if we keep libcrmservice+liblrmd+libstonithd as a separate library, I'm not sure what we'd do with that. I'm leaning to just using pcmk for all our libraries. In any case, who knows when we'll have time for such things, so I'd rather wait until it's on the horizon before making any changes for it.

Reply:
If we are not certain yet, we'd probably better avoid exposing it for now? Probably better put it into pacemaker-fenced.h?

Reply:
Either way is fine. When we eventually merge libraries, I intend to move the public API to new names in the new library and keep the old names in the old library as deprecated wrappers. Pacemaker code would only use the new library/names and the old ones would just be kept for compatibility with external code. But it's a long term idea; I'd start with libcib/libpe_status/libpe_rules first since they'll be easiest.

Reply:
Seems good enough to be put into include/crm/fencing/internal.h, where it can be used by libstonithd, libpacemaker and pacemaker-fenced. Let me know if you don't think it's appropriate.

Reply:
Sure, use a double underbar for internal functions (stonith__)